queries/systemverilog/highlights.scm

```diff
@@ -0,0 +1,528 @@
+; Keywords
+[
+  "begin"
+  "end"
+  "generate"
+  "endgenerate"
+  (module_keyword)
+  "endmodule"
+  "program"
+  "endprogram"
+  "package"
+  "endpackage"
+  "checker"
+  "endchecker"
+  "config"
+  "endconfig"
+  "pure"
+  "virtual"
+  "extends"
+  "implements"
+  "super"
+  (class_item_qualifier)
+  "parameter"
+  "localparam"
+  "defparam"
+  "assign"
+  "typedef"
+  "modport"
+  "fork"
+  "join"
+  "join_none"
+  "join_any"
+  "default"
+  "break"
+  "tagged"
+  "extern"
+  "alias"
+  "posedge"
+  "negedge"
+  "bind"
+  "expect"
+  "type"
+  "void"
+  "coverpoint"
+  "cross"
+  "nettype"
+  "export"
+  "force"
+  "release"
+  "timeunit"
+  "timeprecision"
+  "sequence"
+  "endsequence"
+  "property"
+  "endproperty"
+  "clocking"
+  "endclocking"
+  "iff"
+  "covergroup"
+  "endgroup"
+  "specify"
+  "endspecify"
+  "primitive"
+  "endprimitive"
+  "wait"
+  "wait_order"
+  "const"
+  "constraint"
+  "unique"
+  "do"
+  "genvar"
+  "inside"
+  "rand"
+  "new"
+  "continue"
+  "randc"
+  (unique_priority)
+  (bins_keyword)
+] @keyword
+
+[
+  "class"
+  "endclass"
+  "interface"
+  "endinterface"
+  "enum"
+  "struct"
+  "union"
+] @keyword.type
+
+[
+  "function"
+  "endfunction"
+  "task"
+  "endtask"
+] @keyword.function
+
+"return" @keyword.return
+
+[
+  (always_keyword)
+  "for"
+  "foreach"
+  "repeat"
+  "forever"
+  "initial"
+  "while"
+] @keyword.repeat
+
+[
+  "if"
+  "else"
+  (case_keyword)
+  "endcase"
+] @keyword.conditional
+
+[
+ "?"
+ ":"
+ "#"
+] @keyword.conditional.ternary
```

Comment:
Would you mind only applying these changes to children of `(conditional_expression)`? At least for `":"`, it should be highlighted as `@punctuation.delimiter` broadly (and then get the ternary highlight only when it is part of a ternary statement)